planner/core/planbuilder.go

```diff
@@ -211,6 +212,14 @@ func (info *tableHintInfo) ifPreferHashJoin(tableNames ...*hintTableInfo) bool {
 	return info.matchTableName(tableNames, info.hashJoinTables)
 }
 
+func (info *tableHintInfo) ifPreferOrderedHashJoin(tableNames ...*hintTableInfo) bool {
+	if len(info.orderedHashJoinTables) == 0 {
+		return false
+	}
+	info.matchTableName(tableNames, info.orderedHashJoinTables[1:])
+	return info.matchTableName(tableNames, info.orderedHashJoinTables[:1])
```

Comment:
What's the expected behavior here?
Seems only the first table name will take effect, but we are not warning for the remaining ineffective table names.